### PR TITLE
Fix configuration of Metadata Agent daemon set

### PR DESCRIPTION
**Pull request by kawych**
**What this PR does / why we need it**:
Fixes small errors in Stackdriver Metadata Agent configuration: port number and default version.

**Release note**:
```release-note
Fix port number and default Stackdriver Metadata Agent in daemon set configuration.
```

### CODE REVIEW

**Comment by kawych** [comment]
cc @bmoyles0117 @igorpeshansky

File: cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ spec:
         name: metadata-agent
         ports:
         - containerPort: 8000
-          hostPort: 8000
+          hostPort: 8799
```
Comment (igorpeshansky) [inline cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml]: Don't you also need to change the logging agent configuration to point to this port?

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml (hunk 1 above)
Comment (bmoyles0117) [reply cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml]: Yes, but the logging agent doesn't support the host as a parameter. We should discuss if the port should be part of the same parameter, or a separate parameter entirely. Making it part of METADATA_AGENT_HOSTNAME would be easier, as we could check for the existence of a ":" in the hostname, and append :8000 if not found. WDYT?

Comment (igorpeshansky) [reply cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml]: I'm confused -- the logging agent uses the `metadata_agent_url` parameter, which can definitely include the port. Are you talking about the environment variables used by the container's entrypoint script? If so, then we can add a METADATA_AGENT_PORT variable or rename the current one to METADATA_AGENT_URL.

Comment (bmoyles0117) [reply cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml]: Yeah I'm referencing the container's entrypoint script.

This is where the sed replace occurs
https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/docker/run.sh#L13

And this is what it's replacing "local-metadata-agent.stackdriver.com":8000, not the :8000. Easy enough to update.
https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/docker/Dockerfile#L9

We can discuss the right naming for the env variable, but for now it's not modifiable in any existing docker image.

Comment (kawych) [reply cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml]: Are you OK with this being final configuration for Metadata Agent?

Comment (bmoyles0117) [reply cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml]: I'm ok with this configuration for the Metadata agent. However, in order for these changes to matter we still need to make changes to a few other places.

1. The input source for containers needs to incorporate the local_resource_id label. You can see how did it below.
    The section that this needs to be added in is here: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml#L74

```
$ curl -s https://storage.googleapis.com/stackdriver-container-alpha/configs/stackdriver-agents.yaml | cat -n | grep -C 5 local_resource_id

   205	      <match reform.**>
   206	        @type record_reformer
   207	        enable_ruby true
   208	        <record>
   209	          "logging.googleapis.com/local_resource_id" ${"k8s_container.#{tag_parts[4].split('-')[-1]}"}
   210	        </record>
   212	        tag ${if record['stream'] == 'stderr' then 'raw.kubernetes.stderr' else 'raw.kubernetes.stdout' end}
   213	      </match>
```

2. You need to update the google_cloud output plugin for kubernetes.* to enable the metadata agent.

     https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml#L351

    You can do this by adding these lines. However you manage to get these lines in there, you'll need to ensure that the hostname is set to the `spec.nodeName` from the downwards API. We do this in our `run.sh` as you're familiar with here: https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/docker/run.sh#L15

```
# Enable metadata agent lookups.
enable_metadata_agent true
metadata_agent_url "http://local-metadata-agent.stackdriver.com:8799"
```

**Review by bmoyles0117 — APPROVED** [review]
Please note my comments as we'll need to make changes to the fluentd-gcp kubernetes manifests.

File: cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml (hunk 1 above)
Comment (kawych) [reply cluster/addons/metadata-agent/stackdriver/metadata-agent.yaml]: @crassirostris 
Mik, do you agree with these changes? Do you have this in your plans?

**Comment by jberkus** [comment]
Please update the status of this issue.  Is this PR ready to merge?  Does it require another PR for supporting changes?

**Comment by igorpeshansky** [comment]
This PR is ready to merge. Another PR will be needed to change configurations of other components accordingly.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Comment by kawych** [comment]
@gmarek PTAL

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -158,7 +158,7 @@ ENABLE_METRICS_SERVER="${KUBE_ENABLE_METRICS_SERVER:-true}"
 ENABLE_METADATA_AGENT="${KUBE_ENABLE_METADATA_AGENT:-none}"
 
 # Version tag of metadata agent
-METADATA_AGENT_VERSION="${KUBE_METADATA_AGENT_VERSION:-0.2-0.0.13-5}"
+METADATA_AGENT_VERSION="${KUBE_METADATA_AGENT_VERSION:-0.2-0.0.13-5-watch}"
```
Comment (gmarek) [inline cluster/gce/config-default.sh]: what's that? what's the versioning scheme of metadata agent?

Comment (igorpeshansky) [reply cluster/gce/config-default.sh]: This version will change in a bit once the `watch` branch is merged into `master`. This scheme is: _<Dockerfile version>_-_<package version>_[-_<branch>_]...

**Pull request merged**